File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = Integer.MIN_VALUE;
        int windowSum = 0;
        int start = 0;
        for(int end = 0; end < nums.length; end++){
            windowSum += nums[end];
            while(nums[end] > windowSum){
                windowSum -= nums[start];
                start++;
            }
            maxSum = Math.max(maxSum, windowSum);
        }
        return maxSum;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int maxSum = int.MinValue;
        int windowSum = 0;
        int start = 0;
        for(int end = 0; end < nums.Length; end++){
            windowSum += nums[end];
            maxSum = Math.Max(maxSum, windowSum);
            
            while(windowSum < 0 && start <= end){
                windowSum -= nums[start];
                start++;
            }
        }
        return maxSum;
    }
}